### PR TITLE
feat: Enabled ordering new tracker relationship export endpoint and ordering refactoring [DHIS2-11094]

### PR DESCRIPTION
**Pull request by gnespolino**
This PR enable ordering in NT export endpoint, as well as in Enrollments.
Also I did some minor refactoring to make the code look better.

style: formatting
refactor: moved isLegacy to criteria dto, so that we don't need to pass it around
refactor: moved paging logic for relationship in getList method
feat: added ability for order params to use naming from DTO
feat: added ability for controller's criteria to specify fields on which to allow ordering
feat: added ordering for enrollments controllers

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceQueryParams.java [hunk 1]
```diff
@@ -129,6 +137,11 @@
      */
     private boolean includeDeleted;
 
+    /**
+     * List or order params
+     */
+    private List<OrderParam> order;
```
Comment (lucaCambi77) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceQueryParams.java:143]: Clean code usually suggests, in order to reduce verbosity, to add specific names and no comments, if we want to add an attribute that is a **List or order params** (i think it is also mispelled.. list **of** order params?) we should just call it orderParamsList or similar. Same with other attributes like **pageSize** , do we need a comment to state that is used for page size :) ? ... also considering that there is an effort for less code (see @Accessors)

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java [hunk 2]
```diff
@@ -88,4 +110,44 @@ private boolean isSkipPaging()
...
+    private boolean isAllowed( OrderCriteria orderCriteria )
+    {
+        return getAllowedOrderingFields().contains( orderCriteria.getField() );
+    }
+
+    protected Consumer<? super OrderCriteria> disallowedOrderFieldConsumer()
+    {
+        return orderCriteria -> log.warn( "Ordering by " + orderCriteria.getField() + " is not supported" );
+    };
+
```
Comment (lucaCambi77) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java:143]: we can remove this ; it is also reviewed by SonarQube

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceQueryParams.java (hunk 1 above)
Comment (gnespolino) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceQueryParams.java:143]: yes, comment is not very usefull here, I can remove it.
On the naming, I understand there are better naming, but this is a frontend parameter, so it needs to be "order". Same for pageSize.

Comment (lucaCambi77) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceQueryParams.java:143]: ok i didn't know about front end parameter... yes in my opinion comments should be removed unless explaining something important, but that is my opinion

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java (hunk 2 above)
Comment (gnespolino) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/webapi/controller/event/webrequest/PagingAndSortingCriteriaAdapter.java]: fixed

**Review by lucaCambi77 — APPROVED** [review]

**Review by enricocolasante — APPROVED** [review]

**Pull request merged**